executor/executor.go

```diff
@@ -313,15 +313,24 @@ func (executor *Executor) execute(
 			msg = statusMap[kv1.PodFailed]
 			result = proto.DecisionExecutionStatusFailed
 
-			memoryLimit := container.Resources.Limits.Memory().Value()
-			memoryRequest := container.Resources.Requests.Memory().Value()
-			cpuLimit := container.Resources.Limits.Cpu().MilliValue()
-			cpuRequest := container.Resources.Requests.Cpu().MilliValue()
-
-			*totalResources.Containers[0].Limits.Memory = memoryLimit / 1024 / 1024
-			*totalResources.Containers[0].Requests.Memory = memoryRequest / 1024 / 1024
-			*totalResources.Containers[0].Limits.CPU = cpuLimit
-			*totalResources.Containers[0].Requests.CPU = cpuRequest
+			// handle if requests and limits is null in rollback DEV-2056"
+			if container.Resources.Limits != nil {
```

Comment:
Shouldn't we need to handle the case of if it's nil by failing the job?

Reply:
no because at least one value will be in the recommendation

Reply:
there is no recommendation without both requests and limits

Reply:
@mohamedveron While indeed there is no model in the resource advisor that will generate empty resources object, but what will happen if the logic in the adv were corrupted and somehow generated that recommendation, everything is possible and the logic should be handling any case.

The key point is it important to handle such a case? what's the impact of the resource obj empty and the agent tried to exec it, would it fail? would it stuck in the job forever? if the answer is no in both cases, then it's not much important to handle it now.